tools/querytee/proxy_endpoint_test.go

```diff
@@ -786,3 +786,74 @@ func (b *mockProxyBackend) ForwardRequest(_ context.Context, _ *http.Request, _
 	b.responseIndex++
 	return latency, 200, []byte("{}"), resp, nil
 }
+
+func TestProxyEndpoint_BackendSelection(t *testing.T) {
+	const runCount = 1000
+
+	testCases := map[string]struct {
+		backends                            []ProxyBackendInterface
+		secondaryBackendRequestProportion   float64
+		expectedPreferredOnlySelectionCount int // Out of 1000 runs
```

Comment:
(nit) could calculate this off secondaryBackendRequestProportion

Reply:
I'd prefer to make it explicit - then there's no room for a mistake in the calculation in the test.